payjoin-cli/src/app/v2.rs

```diff
@@ -102,10 +107,17 @@ impl AppTrait for App {
 
 impl App {
     async fn spawn_payjoin_sender(&self, mut req_ctx: RequestContext) -> Result<()> {
-        let res = self.long_poll_post(&mut req_ctx).await?;
-        self.process_pj_response(res)?;
-        self.db.clear_send_session(req_ctx.endpoint())?;
-        Ok(())
+        let mut interrupt = self.interrupt.clone();
+        tokio::select! {
+            res = self.long_poll_post(&mut req_ctx) => {
+                self.process_pj_response(res?)?;
+                self.db.clear_send_session(req_ctx.endpoint())?;
+                Ok(())
+            }
+            _ = interrupt.changed() => {
+                Err(anyhow!("Interrupted. Call `send` with the same arguments to resume this session or `resume` to resume all sessions."))
```

Comment:
Why not `println!` instead of returning an error (like in `resume`)? As it stands it prints: "Error: Interrupted. Call `send` with the same arguments to resume this session or `resume` to resume all sessions."

Reply:
Doh! I was wondering why this looked so wrong. An interrupt isn't an error, it's just a different `Ok` path. Adopted.